test/wasi/args.wast

```diff
@@ -0,0 +1,130 @@
+(module
+  (import "wasi_snapshot_preview1" "args_get" (func $wasi_args_get (param i32 i32) (result i32)))
+  (import "wasi_snapshot_preview1" "args_sizes_get" (func $wasi_args_sizes_get (param i32 i32) (result i32)))
+  (memory 1)
+
+  (export "memory" (memory 0))
+  (data (i32.const 32) "Hello" ) ;; Memory[32] = "Hello,"
+  (data (i32.const 48) "World!" ) ;; Memory[48] = "World!"
+
+  (func (export "check_args")(result i32)
+  (local $i i32)
+  (local $mismatched_char_count i32)
+  (local $argc i32)
+  (local $argv i32)
+  (local $argv_buf i32)
+  (local $value_addr i32)
+  (local $expected_addr i32)
+    ;; Memory[0] = args count
+    ;; Memory[4] = args size in characters
+    i32.const 0
+    local.tee $argc
+    i32.const 4
+    call $wasi_args_sizes_get
+    drop
```

Comment:
could we throw an error on nonzero return code? If you pass a buffer outside the memory area, it should throw error.